### PR TITLE
Fork support for external contributors

### LINKED ISSUE #121: Fork support for external contributors
### Motivation

External contributors working via forks currently face several friction points when using Sherpa CLI:

1. Manual fork creation and remote setup required
2. Need to manually set default repository to upstream
3. Missing fetch operations cause branch creation failures
4. Pull request creation fails due to incorrect remote configuration

This is the current behaviour for the user journey::

```bash
# Current multi-step manual process
gh repo clone InditexTech/gh-sherpa
gh repo fork --remote
gh repo set-default InditexTech/gh-sherpa
git fetch origin
gh sherpa create-branch -i 117
# Still fails on create-pr due to remote issues
```

### Proposed Solution

- Add a `--fork` flag to both `create-branch` and `create-pr` commands that automatically:

1. **Detects repository state** - Check if already in a fork setup
2. **Creates fork if needed** - `gh repo fork --remote` (with user confirmation)
3. **Sets upstream as default** - `gh repo set-default <upstream-repo>`
4. **Fetches from fork** - `git fetch origin`
5. **Proceeds with standard operation** - Create branch/PR with correct remotes

- Additional flag `--fork-name` allows specifying a custom fork organization/user instead of the default behaviour (i.e. user account).
- Add a [configuration parameter](https://github.com/InditexTech/gh-sherpa/blob/main/internal/config/default-config.yml) to overwrite the default fork organization for convenience 

Examples:

```bash
# One-command fork setup and branch creation
gh sherpa create-branch -i 32 --fork

# One-command fork setup and PR creation
gh sherpa create-pr -i 32 --fork

# Custom fork organization
gh sherpa create-branch -i 45 --fork --fork-name=MyOrg/gh-sherpa
```

#### Use Cases

- First-time Fork Setup:

```bash
❯ gh sherpa create-branch -i 32 --fork
=> Detecting repository setup...
=> No fork detected. Creating fork danielfn/gh-sherpa...
✓ Fork created successfully
=> Setting up remotes (origin: fork, upstream: original)...
=> Setting default repository to upstream...
=> Fetching branches from fork...
=> Creating branch bugfix/GH-32-fix-link-to-cla...
✓ Ready to start working on issue #32!
```

- Subsequent Usage:
```bash
❯ gh sherpa create-branch -i 45 --fork
=> Fork already configured, creating branch...
=> Creating branch feature/GH-45-new-feature...
✓ Ready to start working on issue #45!
```

#### Implementation considerations

Detection logic:
- Check if current repository is a fork
- Verify remote configuration (origin = fork, upstream = original)
- Validate default repository setting

Safety Features:
- Confirmation prompt before creating fork (unless `--yes` flag)
- Verbose logging of all automated steps

Edge Cases:
- Repository already forked by user
- No permission to fork
- Network connectivity issues
- Invalid repository state

Documentation Impact:
- Update `USAGE.md` with `--fork` parameter documentation
- Include troubleshooting guide for common fork scenarios
- Update examples to show fork-based workflows
- Add configuration parameter for default fork organization

### PR DESCRIPTION
**Pull request by carlosmmesa**
Closes #121

Command Examples:
- Create branch "regular execution":
  ``` 
  ➜  gh-sherpa git:(feature/GH-121-fork-support-for-external-c) ./gh-sherpa create-branch -i 118 --fork

  => Running create-branch command in Sherpa.

  INFO: Detecting repository setup...
  No fork detected. Creating fork...
  ? Do you want to create a fork and configure it for development? Yes
  ✓ Fork created successfully
  Setting up remotes (origin: fork, upstream: original)...
  Setting default repository to upstream...
  Fetching branches from fork...
  ✓ Ready to start working on issue #118!
  ? Label 'kind/bugfix' found. What type of branch name do you want to create? bugfix
  ? Please, write one additional description (optional). Truncate to 28 chars: fix-link-to-cla

  A new local branch named bugfix/GH-118-fix-link-to-cla is going to be created
  ? Do you want to continue? Yes
  A local branch named bugfix/GH-118-fix-link-to-cla has been created!
  ➜  gh-sherpa git:(bugfix/GH-118-fix-link-to-cla)
   ```
- Denying the fork creation:
  ```
  ➜  gh-sherpa git:(feature/GH-121-fork-support-for-external-c) ./gh-sherpa create-branch -i 103 --fork

  => Running create-branch command in Sherpa.

  INFO: Detecting repository setup...
  No fork detected. Creating fork...
  ? Do you want to create a fork and configure it for development? No
  ERROR: fork creation cancelled by user
  ```
- Create another branch after the fork is created:
  ```
  ➜  gh-sherpa git:(feature/GH-121-fork-support-for-external-c) ✗ ./gh-sherpa create-branch -i 118 --fork

  => Running create-branch command in Sherpa.

  INFO: Detecting repository setup...
  Fork already configured, creating branch...

  A new local branch named bugfix/GH-118-fix-link-to-cla is going to be created
  A local branch named bugfix/GH-118-fix-link-to-cla has been created!
  ```
- Create a branch in a custom fork:
  ```
  ➜  gh-sherpa-test git:(feature/GH-121-fork-support-for-external-c) ./gh-sherpa create-branch -i 103 --fork --fork-name=carlosmmesa-test-org/gh-sherpa

  => Running create-branch command in Sherpa.

  INFO: Detecting repository setup...
  No fork detected. Creating fork carlosmmesa-test-org/gh-sherpa...
  ? Do you want to create a fork and configure it for development? Yes
  ✓ Fork created successfully
  Setting up remotes (origin: fork, upstream: original)...
  Setting default repository to upstream...
  Fetching branches from fork...
  ✓ Ready to start working on issue #103!
  ? Label 'kind/feature' found. What type of branch name do you want to create?  [Use arrows to move, type to filter]
  ...
  ```
- Create PR "regular execution":
  ```
  ➜  gh-sherpa-test git:(feature/GH-121-fork-support-for-external-c) ./gh-sherpa create-pr -i 103 --fork

  => Running create-pr command in Sherpa.

  INFO: Detecting repository setup...
  Fork already configured, creating branch...
  ? Label 'kind/feature' found. What type of branch name do you want to create? feature
  ? Please, write one additional description (optional). Truncate to 27 chars: add-cli-flag-for-description

  A new pull request is going to be created from feature/GH-103-add-cli-flag-for-descriptio to main branch
  ? Do you want to continue? Yes

  The pull request https://github.com/InditexTech/gh-sherpa/pull/125 have been created!
  You are now working on the branch feature/GH-103-add-cli-flag-for-descriptio
  ```
- Create a PR after a fork has been created in a custom fork scenario (it is created in the custom fork):
  ```
  ➜  test-fork git:(bugfix/GH-10-test-bug) ../gh-sherpa-cli create-pr -i 1 --fork                          

  => Running create-pr command in Sherpa.

  INFO: Detecting repository setup...
  Fork already configured, creating branch...
  ? Label 'kind/feature' found. What type of branch name do you want to create? feature
  ? Please, write one additional description (optional). Truncate to 29 chars: test-feature

  A new pull request is going to be created from feature/GH-1-test-feature to main branch
  ? Do you want to continue? Yes

  The pull request https://github.com/carlosmmesa/test-fork/pull/13 have been created!
  You are now working on the branch feature/GH-1-test-feature
  ```
- Create a PR in a custom fork, extracted from the Sherpa configuration:
  ```
  ➜  test-fork git:(main) ./gh-sherpa create-pr -i 10 --fork

  => Running create-pr command in Sherpa.

  INFO: Detecting repository setup...
  No fork detected. Creating fork carlosmmesa-test-org/test-fork...
  ? Do you want to create a fork and configure it for development? Yes
  ✓ Fork created successfully
  Setting up remotes (origin: fork, upstream: original)...
  Fetching branches from fork...
  ✓ Ready to start working on the pull request!
  ? Label 'kind/bugfix' found. What type of branch name do you want to create? bugfix
  ? Please, write one additional description (optional). Truncate to 29 chars: test-bug

  A new pull request is going to be created from bugfix/GH-10-test-bug to main branch
  ? Do you want to continue? Yes

  The pull request https://github.com/carlosmmesa/test-fork/pull/16 have been created!
  You are now working on the branch bugfix/GH-10-test-bug
  ```
- Edge cases:
  - Repository already forked by user:
    ```
    ➜  gh-sherpa-test git:(main) ./gh-sherpa create-pr -i 118 --fork --fork-name=carlosmmesa-test-org/gh-sherpa

    => Running create-pr command in Sherpa.

    INFO: Detecting repository setup...
    Fork carlosmmesa-test-org/gh-sherpa already exists, configuring for use...
    Setting up remotes (origin: fork, upstream: original)...
    Setting default repository to upstream...
    Fetching branches from fork...
    ? Label 'kind/bugfix' found. What type of branch name do you want to create?  [Use arrows to move, type to filter]
    > bugfix
    ...
    ```
  - No permission to fork
    ```
    ➜  gh-sherpa git:(feature/GH-121-fork-support-for-external-c) ✗ ./gh-sherpa create-branch -i 103 --fork --fork-name=m4rii0/gh-sherpa

    => Running create-branch command in Sherpa.

    INFO: Detecting repository setup...
    No fork detected. Creating fork m4rii0/gh-sherpa...
    ? Do you want to create a fork and configure it for development? Yes
    ERROR: failed to create fork: error creating fork: failed to run GitHub CLI command (gh execution failed: exit status 1)

    Details:
    failed to fork: HTTP 403: Must have admin rights to Repository. (https://api.github.com/repos/InditexTech/gh-sherpa/forks)
    ```
  - Try to create a PR directly after creating a fork without the `--fork` label (works as the fork is already configured):
    ```
    ➜  gh-sherpa-test git:(feature/GH-121-fork-support-for-external-c) ./gh-sherpa create-pr -i 103

    => Running create-pr command in Sherpa.

    ? Label 'kind/feature' found. What type of branch name do you want to create? feature
    ? Please, write one additional description (optional). Truncate to 27 chars: add-cli-flag-for-description

    A new pull request is going to be created from feature/GH-103-add-cli-flag-for-descriptio to main branch
    ? Do you want to continue? Yes

    The pull request https://github.com/InditexTech/gh-sherpa/pull/126 have been created!
    You are now working on the branch feature/GH-103-add-cli-flag-for-descriptio
    ```
  - Invalid repository state:
    - Fork already configured, and try to fork again in another fork setup:
      ```
      ➜  gh-sherpa git:(feature/GH-121-fork-support-for-external-c) ✗ ./gh-sherpa create-branch -i 118 --fork --fork-name=MyOrg/gh-sherpa

      => Running create-branch command in Sherpa.

      INFO: Detecting repository setup...
      ERROR: fork mismatch: repository is already configured with fork 'carlosmmesa/gh-sherpa', but you requested 'MyOrg/gh-sherpa'. Please use the existing fork or reconfigure the repository
      ```
    - Fork configured but repository doesn't exist:
      ```
      ➜  gh-sherpa git:(feature/GH-121-fork-support-for-external-c) ✗ ./gh-sherpa create-pr -i 118 --fork

      => Running create-pr command in Sherpa.

      INFO: Detecting repository setup...
      Fork already configured, creating branch...
      ERROR: could not fetch the branch main: remote: Repository not found.
      fatal: repository 'https://github.com/carlosmmesa/gh-sherpa.git/' not found
      ```

### CODE REVIEW

File: internal/fork/types.go
Comment (danielfn) [inline internal/fork/types.go:1]: This file looks like a good candidate to be into the `domain` package

File: internal/utils/git_url.go [hunk 1]
```diff
@@ -0,0 +1,27 @@
+package utils
+
+import (
+	"fmt"
+	"strings"
+)
+
+// ExtractRepoFromURL extracts owner/repo from a git URL
+func ExtractRepoFromURL(url string) string {
+	if strings.Contains(url, "github.com") {
```
[the remark this answers is not in the document]
Comment (carlosmmesa) [reply internal/utils/git_url.go]: Done in [6adb860](https://github.com/InditexTech/gh-sherpa/pull/122/commits/6adb860cc45a9589f48fffd72dd0652d0f4cb355)

File: internal/fork/types.go
Comment (carlosmmesa) [reply internal/fork/types.go:1]: Done in [fc9dd47](https://github.com/InditexTech/gh-sherpa/pull/122/commits/fc9dd4779af41206c56fd0c95a20f35abee2fa11)!

File: cmd/common/fork.go [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+		return nil
+	}
+
+	if result.ForkCreated {
+		if messageType == "issue" && issueValue != "" {
+			fmt.Printf("✓ Ready to start working on issue #%s!\n", issueValue)
+		} else {
+			fmt.Printf("✓ Ready to start working on the %s!\n", messageType)
+		}
+	}
```
Comment (danielfn) [inline cmd/common/fork.go:53]: I think this message doesn't make sense at this point, since the only thing created is the fork (the branch or PR is created afterwards). I would remove this block altogether, since the proper messaging is handled in the use cases.

**Review by danielfn — CHANGES_REQUESTED** [review]

File: cmd/common/fork.go (hunk 2 above)
Comment (carlosmmesa) [reply cmd/common/fork.go]: Done in [c8c9d96](https://github.com/InditexTech/gh-sherpa/pull/122/commits/c8c9d964406df501fb598819ce879c820c473194)!

**Review by danielfn — APPROVED** [review]
LGTM!
